### PR TITLE
StorageSync Private Link changes

### PR DESCRIPTION
**Pull request by ankushbindlish2**
I deleted my repo by mistake. Here is the old PR https://github.com/Azure/azure-rest-api-specs/pull/8942

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by akning-ms** [comment]
reassign to @weidongxu-microsoft as he is reviewer for #8942 , based on our offline discussing, it is not same PR as #8942.  @ankushbindlish2  pls fix above validation failure refer to below document:
https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/ci-fix.md

File: specification/storagesync/resource-manager/Microsoft.StorageSync/preview/2020-03-01/storagesync.json [hunk 1]
```diff
@@ -162,13 +162,39 @@
               "$ref": "#/definitions/StorageSyncService"
             }
           },
+          "202": {
```
Comment (weidongxu-microsoft) [inline specification/storagesync/resource-manager/Microsoft.StorageSync/preview/2020-03-01/storagesync.json:165]: Continue from previous PR.

This appears not consistent with most common services.
https://github.com/Azure/azure-rest-api-specs/blob/master/specification/compute/resource-manager/Microsoft.Compute/stable/2019-12-01/compute.json#L915-L934

File: specification/storagesync/resource-manager/Microsoft.StorageSync/preview/2020-03-01/storagesync.json [hunk 2]
```diff
@@ -278,6 +304,31 @@
...
+              "x-ms-request-id": {
+                "description": "Request id",
+                "type": "string"
+              },
+              "x-ms-correlation-request-id": {
+                "description": "correlation request id",
+                "type": "string"
+              }
+            }
+          },
```
Comment (weidongxu-microsoft) [inline specification/storagesync/resource-manager/Microsoft.StorageSync/preview/2020-03-01/storagesync.json:331]: And it seems not consistent within this json as well.

Does this one need a `"x-ms-long-running-operation": true` after `default`?

**Comment by weidongxu-microsoft** [comment]
@ankushbindlish2 

I will not block you on this. If you double confirm that the "202" response is OK, then it is fine.

Please check BreakingChange CI and confirm these are intended.

Ping me and akning-ms when you want to merge the PR.

**Review by weidongxu-microsoft — APPROVED** [review]

**Pull request merged**